File: service/src/mocks/subscription.js
const data = [
  {
    track: true,
    distanceToNotifier: 1000,
    latitude: 10.0,
    longitude: 10.0,
  },
  {
    track: false,
    distanceToNotifier: 500,
    latitude: -4.561288, // -4.560945, -37.770139 - 139 meters
    longitude: -37.768931,
  },
  {
    track: false,
    distanceToNotifier: 100,
    latitude: 100.0,
    longitude: 100.0,
  },
];

function getRandomInt(min, max) {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min)) + min;
}

function randomSelect(items = []) {
  return items[getRandomInt(0, items.length - 1)];
}

function makeData(mobiles = [], users = []) {
  return data.map((values) => ({
    ...values,
    mobileId: randomSelect(mobiles),
    userId: randomSelect(users),
  }));
}

module.exports = {
  makeData,
};

File: demo_client/lib/model/subscription.dart
class SubscriptionModel {
  SubscriptionModel(
    this.track,
    this.userId,
    this.mobileId,
    this.latitude,
    this.longitude,
    this.distanceToNotifier,
  );

  bool? track;
  String? userId;
  String? mobileId;
  double? latitude;
  double? longitude;
  double? distanceToNotifier;

  SubscriptionModel.fromJson(Map<String, dynamic> json)
      : mobileId = json['mobileId'],
        track = json['track'],
        userId = json['userId'],
        latitude = json['latitude'],
        longitude = json['longitude'],
        distanceToNotifier = json['distanceToNotifier'];

  Map<String, dynamic> toJson() => {
        'mobileId': mobileId,
        'track': track,
        'userId': userId,
        'latitude': latitude,
        'longitude': longitude,
        'distanceToNotifier': distanceToNotifier,
      };
}
